test/jit/test_cuda.py

```diff
@@ -0,0 +1,410 @@
+        @torch.jit.script
+        def test_event():
+            device_index = torch.cuda.current_device()
+            stream = torch.cuda.current_stream(device_index)
+            event = torch.jit.cuda.Event(True, False, False)
+            is_true_event_query = event.query()
+            start_event = torch.jit.cuda.Event(True, False, False)
+            stream.record_event(start_event)
+            tensor1 = torch.rand(1000000000, 1000000000, device = "cuda")
+            tensor2 = torch.mm(tensor1, tensor1).to("cuda")
+            stream.record_event(event)
+            is_false_event_query = event.query()
+            event.synchronize()
+            is_again_true_event_query = event.query()
+
+            if not (is_true_event_query and is_again_true_event_query):
+                return -1.0
+            return start_event.elapsed_time(event)
+
+        self.assertGreater(test_event(), 0)
+
+        @unittest.skipIf(not TEST_LARGE_TENSOR, "not enough memory")
+        @torch.jit.script
+        def test_stream_synchronize() -> float:
+            device_index = torch.cuda.current_device()
+            s = torch.jit.cuda.Stream(device_index, 0)
+            e_tik = torch.jit.cuda.Event(True, False, False)
+            e_tok = torch.jit.cuda.Event(True, False, False)
+
+            e_tik.record(s)
+            tensor1 = torch.rand(1000000000, 1000000000, device = "cuda")
+            with torch.jit.cuda.stream(s):
+                tensor2 = torch.mm(tensor1, tensor1).to("cuda")
+            s.synchronize()
+            e_tok.record(s)
+
+            if not s.query():
+                return -1.0
+
+            # not necessary to check e_tik and e_tok, as elapsed_time would throw
+            # exception if otherwise.
+            return e_tik.elapsed_time(e_tok)
+        self.assertGreater(test_stream_synchronize(), 0)
+
+        @unittest.skipIf(not TEST_LARGE_TENSOR, "not enough memory")
+        @torch.jit.script
+        def test_event_synchronize() -> float:
+            device_index = torch.cuda.current_device()
+            s = torch.jit.cuda.Stream(device_index, 0)
+            e_tik = torch.jit.cuda.Event(True, False, False)
+            e_tok = torch.jit.cuda.Event(True, False, False)
+
+            e_tik.record(s)
+            tensor1 = torch.rand(1000000000, 1000000000, device = "cuda")
+            with torch.jit.cuda.stream(s):
+                tensor = torch.mm(tensor1, tensor1).to("cuda")
+            is_false = s.query()
+            e_tok.synchronize()
+            s.record_event(e_tok)
+
+            if not s.query():
+                return -1.0
+
+            # not necessary to check e_tik and e_tok, as elapsed_time would throw
+            # exception if otherwise.
+            return e_tik.elapsed_time(e_tok)
+
+        self.assertGreater(test_event_synchronize(), 0)
+
+        @unittest.skipIf(not TEST_LARGE_TENSOR, "not enough memory")
+        @torch.jit.script
+        def test_event_wait() -> float:
```

Comment:
I think it would be a good idea to add a more sophisticated test that uses the non-blocking version of `wait` to make ops on stream A wait for ops on stream B to finish *and* uses the results of the ops from stream A.

Reply:
Agreed, I'll add.

Reply:
Have added a comment on Line 227